### PR TITLE
New structure of codalab, codalabworker modules

### PR DESCRIPTION
**Pull request by maxwang7**
Creates `codalabworker` as a separate module, and changes the name of `codalab-cli` to `codalab`. Also includes the `cl-worker` script as an executable command.

### CODE REVIEW

**Review by kashizui — APPROVED** [review]
LGTM modulo the circleCI errors and merge conflicts.

Also, any specific reason why `codalabworker` is nested inside another directory called `worker`? Since the `codalab` package sits directly at the base of the repo as well.

**Comment by percyliang** [comment]
Yeah, seems better if codalabworker were at the top-level.
Given the number of changes here, let's really test thoroughly before
deployment.

File: worker/setup.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        with open(requirements_file_path) as requirements_file:
+            for line in requirements_file:
+                if line[0:2] != '-r' and line.find('git') == -1 and line.find('codalabworker') == -1:
+                    requirements.append(line)
+    return requirements
+
+setup(name='codalabworker',
+    version='0.0.4',
+    description='Worker for CodaLab, a platform for reproducible computation',
+    long_description='To use your own hardware in CodaLab Worksheets, visit https://github.com/codalab/codalab-worksheets/wiki/Execution#running-your-own-workervisit. You can find the code at https://github.com/codalab/codalab-cli.',
```
Comment (percyliang) [inline worker/setup.py]: typo: running-your-own-workervisit

**Comment by fabeschan** [comment]
lgtm, but make sure changes passes tests and also that codalab-deployment/ansible can handle the `worker -> codalabworker` name change

**Comment by maxwang7** [comment]
Thank you everyone for the fast reviews!

I had to nest `codalabworker` under another directory because `setuptools` must use a file named `setup.py`. My original didn't nest directories and I couldn't find a way around this problem. I agree that it's not preferable.  

And sounds good @fabeschan, should I just try deploying to dev with this branch?

**Comment by maxwang7** [comment]
Passing CI tests!!!

**Comment by maxwang7** [comment]
Fixes codalab/codalab-worksheets#314

**Pull request merged**